mvp_onboarding.client/src/components/CustomerTable.jsx

```diff
@@ -0,0 +1,354 @@
+  }
+
+  useEffect(() => {
+    fetchCustomers(currentPage, pageSize, sortColumn, sortDirection);
+  }, [currentPage, pageSize, sortColumn, sortDirection]);
+
+  useEffect(() => {
+    setCreateIsDisabled(invalidSelectedCustomer);
+    setEditIsDisabled(invalidSelectedCustomer);
+  }, [selectedAddress, selectedName]);
+
+  const fetchCustomers = async (currentPage, pageSize, sortColumn, sortDirection) => {
+    try {
+      const data = await getCustomers(currentPage, pageSize, sortColumn, sortDirection);
+      if (data?.items?.length > 0) {
+        setCustomers(data?.items);
+        setTotalCount(data?.totalCount);
+      }
+    } catch (error) {
+      console.error("Failed to fetch customers", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+  const handleDelete = async () => {
+    if (selectedCustomer) {
+      try {
+        await deleteCustomer(selectedCustomer?.id);
+        setCustomers(customers.filter((customer) => customer.id !== selectedCustomer?.id));
+        setDeleteOpen(false);
+      }
+      catch (error) {
+        console.error("Failed to delete customer", error);
+      }
+    }
+  };
+
+  const handleEditSubmit = async () => {
+    let newCustomer = {
+      id: selectedCustomer.id,
+      name: selectedName,
+      address: selectedAddress
+    };
+    if (newCustomer) {
+      try {
+        await updateCustomer(selectedCustomer.id, newCustomer);
+        fetchCustomers(currentPage, pageSize, sortColumn, sortDirection);
+      }
+      catch (error) {
+        console.error("Failed to update customer", error);
+      }
+      setEditOpen(false);
+    }
+
+  }
+  const handleNewSubmit = async () => {
+    let newCustomer = {
+      id: "0",
+      name: selectedName,
+      address: selectedAddress
+    };
+    setSelectedCustomer(newCustomer);
+    if (newCustomer) {
+      try {
+        await createCustomer(newCustomer);
+        fetchCustomers(currentPage, pageSize, sortColumn, sortDirection);
+      }
+      catch (error) {
+        console.error("Failed to add new customer", error);
+      }
+      setNewOpen(false);
+    }
+  };
+
+  const confirmDelete = (customer) => {
+    setSelectedCustomer(customer);
+    setDeleteOpen(true);
+  };
+
+  const confirmEdit = (customer) => {
+    setSelectedCustomer(customer);
+    setSelectedName(customer.name);
+    setSelectedAddress(customer.address);
+    setEditOpen(true);
+  };
+
+  const confirmNewSubmit = () => {
+    const customer = { id: "0", name: "", address: "" };
+    setSelectedName("");
+    setSelectedAddress("");
+    setSelectedCustomer(customer);
+    setNewOpen(true);
+  };
+
+
+  const handleNameChange = (event) => {
+    setSelectedName(event.target.value);
+  };
+
+  const handleAddressChange = (event) => {
+    setSelectedAddress(event.target.value);
+  };
+
+  const invalidSelectedCustomer = () => {
+    if (selectedName === "") {
+      return true;
+    }
+    if (selectedAddress === "") {
+      return true;
+    }
```

Comment:
You can simplify this condition. Will **selectedName === "" && selectedAddress === ""** works?